libs/libc/string/lib_strnlen.c

```diff
@@ -30,7 +30,8 @@
  * Public Functions
  ****************************************************************************/
 
-#ifndef CONFIG_LIBC_ARCH_STRNLEN
+#if !defined(CONFIG_LIBC_ARCH_STRNLEN) && \
```

Comment:
why not remove CONFIG_LIBC_ARCH_STRNLEN check directly? lib_strnlen.c is already skipped in Makefile if CONFIG_LIBC_ARCH_STRNLEN equals y, which make the check redundancy.

Reply:
Yes, it's redundant. I just kept the old logic and added the new logic. It isn't directly related to this PR, so I preferred to keep it. What do you think about it?

Reply:
I created a previous commit to remove redundant code. Could you please take a look, @xiaoxiang781216 ?